demos/ota/ota_demo_core_mqtt/ota_demo_core_mqtt.c

```diff
@@ -0,0 +1,1354 @@
+
+        /* Nothing special to do. The OTA agent handles it. */
+    }
+    else if( event == OtaJobEventStartTest )
+    {
+        /* This demo just accepts the image since it was a good OTA update and networking
+         * and services are all working (or we would not have made it this far). If this
+         * were some custom device that wants to test other things before calling it OK,
+         * this would be the place to kick off those tests before calling OTA_SetImageState()
+         * with the final result of either accepted or rejected. */
+
+        LogInfo( ( "Received OtaJobEventStartTest callback from OTA Agent." ) );
+        err = OTA_SetImageState( OtaImageStateAccepted );
+
+        if( err != OtaErrNone )
+        {
+            LogError( ( " Error! Failed to set image state as accepted." ) );
+        }
+    }
+    else if( event == OtaJobEventProcessed )
+    {
+        LogDebug( ( "Received OtaJobEventProcessed callback from OTA Agent." ) );
+
+        if( pData != NULL )
+        {
+            otaEventBufferFree( ( OtaEventData_t * ) pData );
+        }
+    }
+}
+
+/*-----------------------------------------------------------*/
+
+static void mqttJobCallback( MQTTContext_t * pContext,
+                             MQTTPublishInfo_t * pPublishInfo )
+{
+    assert( pPublishInfo != NULL );
+    assert( pContext != NULL );
+
+    OtaEventData_t * pData;
+    OtaEventMsg_t eventMsg = { 0 };
+
+    LogInfo( ( "Received job message callback, size %ld.\n\n", pPublishInfo->payloadLength ) );
+
+    pData = otaEventBufferGet();
+
+    if( pData != NULL )
+    {
+        memcpy( pData->data, pPublishInfo->pPayload, pPublishInfo->payloadLength );
+        pData->dataLength = pPublishInfo->payloadLength;
+        eventMsg.eventId = OtaAgentEventReceivedJobDocument;
+        eventMsg.pEventData = pData;
+
+        /* Send job document received event. */
+        OTA_SignalEvent( &eventMsg );
+    }
+    else
+    {
+        LogError( ( "Error: No OTA data buffers available.\r\n" ) );
+    }
+}
+
+/*-----------------------------------------------------------*/
+
+static void mqttDataCallback( MQTTContext_t * pContext,
+                              MQTTPublishInfo_t * pPublishInfo )
+{
+    assert( pPublishInfo != NULL );
+    assert( pContext != NULL );
+
+    OtaEventData_t * pData;
+    OtaEventMsg_t eventMsg = { 0 };
+
+    LogInfo( ( "Received data message callback, size %zu.\n\n", pPublishInfo->payloadLength ) );
+
+    pData = otaEventBufferGet();
+
+    if( pData != NULL )
+    {
+        memcpy( pData->data, pPublishInfo->pPayload, pPublishInfo->payloadLength );
+        pData->dataLength = pPublishInfo->payloadLength;
+        eventMsg.eventId = OtaAgentEventReceivedFileBlock;
+        eventMsg.pEventData = pData;
+
+        /* Send job document received event. */
+        OTA_SignalEvent( &eventMsg );
+    }
+    else
+    {
+        LogError( ( "Error: No OTA data buffers available.\r\n" ) );
```

Comment:
```suggestion
        LogError( ( "No OTA data buffers available.\r\n" ) );
```

Reply:
Addressed in https://github.com/aws/aws-iot-device-sdk-embedded-C/pull/1418/commits/726a7d066006ec340f26308d37075a414069f56b